### PR TITLE
Move operations to Field

### PR DESCRIPTION
**Pull request by matthiasgoergens**
Implement the TODOs and move `from_noncanonical_u64` and `from_noncanonical_i64` from `Field64` to `Field`.

### CODE REVIEW

**Comment by matthiasgoergens** [comment]
Compare https://github.com/mir-protocol/plonky2/pull/1092

File: field/src/types.rs [hunk 1]
```diff
@@ -341,6 +341,26 @@ pub trait Field:
...
+        Self::from_noncanonical_u128(n as u128)
+    }
+
+    /// Returns `n` as an element of this field.
+    ///
+    /// Implemented by default via `from_noncanonical_u128` and case analysis.
+    /// Override, if you have a faster implementation for your field.
+    fn from_noncanonical_i64(n: i64) -> Self {
+        if n < 0 {
+            -Self::from_noncanonical_u128((n as i128).unsigned_abs())
```
Comment (vivekvpandya) [inline field/src/types.rs:358]: sorry , how to understand this negation?

Comment (matthiasgoergens) [reply field/src/types.rs]: I removed the default implementation.

**Comment by matthiasgoergens** [comment]
I merged this upstream already.  So we can take it from there.

**Pull request closed without merging**